specification/hybridcompute/resource-manager/Microsoft.HybridCompute/preview/2022-08-11-preview/HybridCompute.json

```diff
@@ -0,0 +1,1627 @@
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The location of the Extension being received."
+          },
+          {
+            "name": "publisher",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The publisher of the Extension being received."
+          },
+          {
+            "name": "extensionType",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The extensionType of the Extension being received."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ExtensionValueListResult"
+            }
+          },
+          "default": {
+            "description": "Error in retrieving extension list",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "GET a list of extensions": {
+            "$ref": "./examples/ExtensionMetadata_List.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.HybridCompute/operations": {
+      "get": {
+        "tags": [
+          "operations"
+        ],
+        "operationId": "Operations_List",
+        "description": "Gets a list of hybrid compute operations.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List Hybrid Compute Provider Operations": {
+            "$ref": "./examples/Operations_List.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ExtensionValueListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+            "$ref": "#/definitions/ExtensionValue"
+          },
+          "description": "The list of extension metadata",
+          "x-ms-identifiers": []
+        }
+      },
+      "description": "The List Extension Metadata response."
+    },
+    "ExtensionValue": {
```

Comment:
>ExtensionValue [](http://example.com/codeflow?start=5&length=14)

Why did you replace the ProxyResource definition with id and location? You also need "type" as a minimum, but using the common type definition is much better. If you aren't ready for this resource to have systemData, you can use ProxyResource from v2 common types.

Reply:
I am a little confused as to what you mean by replacing the proxyresource definition, can you elaborate on the issue a little further and tell me what needs to get done to resolve it?

Reply:
At commit 12 this definition included a $ref to v1 common types definition ProxyResource which has the required top-level properties id, name, and type. Now that $ref is gone, and an id and name property are added in this definition. Recommendation to fix it is to add a $ref to the ProxyResource definition in v3 common types and remove the id and name from this definition.

Reply:
@mentat9 - Thank you Chris.
Updated to include ProxyResource v3 common-type.
Lmk if this looks good